src/main/java/net/sf/jabref/importer/fileformat/BibTeXMLImporter.java

```diff
@@ -78,33 +102,539 @@ public ParserResult importDatabase(BufferedReader reader) throws IOException {
+                }
+                if (entry.getProceedings() != null) {
+                    bibEntry.setType("proceedings");
+                    parseProceedings(entry.getProceedings(), fields);
+                }
+                if (entry.getTechreport() != null) {
+                    bibEntry.setType("techreport");
+                    parseTechReport(entry.getTechreport(), fields);
+                }
+                if (entry.getUnpublished() != null) {
+                    bibEntry.setType("unpublished");
+                    parseUnpublished(entry.getUnpublished(), fields);
+                }
+                if (entry.getId() != null) {
+                    bibEntry.setCiteKey(entry.getId());
+                }
+                bibEntry.setField(fields);
+                bibItems.add(bibEntry);
+            }
+        } catch (JAXBException | XMLStreamException e) {
             LOGGER.error("Error with XML parser configuration", e);
             return ParserResult.fromErrorMessage(e.getLocalizedMessage());
-        } catch (org.xml.sax.SAXException e) {
-            LOGGER.error("Error during XML parsing", e);
-            return ParserResult.fromErrorMessage(e.getLocalizedMessage());
-        } catch (IOException e) {
-            LOGGER.error("Error during file import", e);
-            return ParserResult.fromErrorMessage(e.getLocalizedMessage());
         }
         return new ParserResult(bibItems);
     }
 
+    private void parseUnpublished(Unpublished unpublished, HashMap<String, String> fields) {
+        putIfValueNotNull(fields, FieldName.ABSTRACT, unpublished.getAbstract());
+        putIfValueNotNull(fields, "affiliation", unpublished.getAffiliation());
+        putIfValueNotNull(fields, FieldName.AUTHOR, unpublished.getAuthor());
+        putIfValueNotNull(fields, "category", unpublished.getCategory());
+        putIfValueNotNull(fields, "copyright", unpublished.getCopyright());
+        putIfValueNotNull(fields, FieldName.CROSSREF, unpublished.getCrossref());
+        putIfValueNotNull(fields, FieldName.ISBN, unpublished.getIsbn());
+        putIfValueNotNull(fields, FieldName.ISSN, unpublished.getIssn());
+        putIfValueNotNull(fields, FieldName.KEY, unpublished.getKey());
+        putIfValueNotNull(fields, FieldName.KEYWORDS, unpublished.getKeywords());
+        putIfValueNotNull(fields, FieldName.LANGUAGE, unpublished.getLanguage());
+        putIfValueNotNull(fields, "lccn", unpublished.getLccn());
+        putIfValueNotNull(fields, FieldName.LOCATION, unpublished.getLocation());
+        putIfValueNotNull(fields, FieldName.MONTH, unpublished.getMonth());
+        putIfValueNotNull(fields, "mrn-number", unpublished.getMrnumber());
+        putIfValueNotNull(fields, FieldName.NOTE, unpublished.getNote());
+        putIfValueNotNull(fields, FieldName.TITLE, unpublished.getTitle());
+        putIfValueNotNull(fields, FieldName.URL, unpublished.getUrl());
```

Comment:
There is quite a lot of redundant code, please extract the code snippets with the same code to a new method. You can create your methods after the same schema as the xsd:group elements are build in the file bibtexml.xsd.

For example extract all fields that are specified in the "user" group starting at line 77 in bibtexml.xsd and create a method `private void putIfUserNotNull()`.